### ISSUE TITLE
Rule proposal: `prefer-error-cause`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1342]
When throwing a new error in `try/catch` or `Promise#catch`, should add the old error as new error's cause property.

- ["Error Cause" proposal](https://github.com/tc39/proposal-error-cause) 

## Fail

```js
try {} catch {
	throw new Error('oops');
}
```

```js
// (I'm not sure about this)
try {} catch (error) {
	error.message = 'oops';
	throw error;
}
```

```js
promise.catch(() => {
	throw new Error('oops');
})
```

```js
// Same as above
promise.then(null, () => {
	throw new Error('oops');
})
```

## Pass

```js
try {} catch (error) {
	throw new Error('oops', {cause: error});
}
```

```js
promise.catch((error) => {
	throw new Error('oops', {cause: error});
})
```

**Comment by sindresorhus** [comment]
This is now accepted.

**Comment by fregante** [comment]
It could also catch the same pattern in extended errors:


### Fail

```js
class ExtendedError extends Error {
	constructor(message: string, error: Error) {
		super(message);
		this.innerError = error;
	}
}
```

### Pass


```js
class ExtendedError extends Error {
	constructor(message, {cause}) {
		super(message, {cause});
	}
}
// or

class ExtendedError extends Error {}
```

**Comment by sindresorhus** [comment]
If anyone wants to work on this, see the initial attempt and feedback in https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1793.

**Comment by fisker** [comment]
Closing since it has already been implemented.

Note: `Promise.catch` is not included https://github.com/eslint/eslint/issues/19844#issuecomment-2980656392
